src/FsCodec/StreamName.fs

```diff
@@ -15,116 +15,88 @@ and [<Measure>] streamName
+        /// The <c>{streamId}</c> segment is permitted to include embedded '-' (dash) characters
+        /// Yields <c>NotCategorized</c> if it does not adhere to that form.</summary>
+        let (|Categorized|NotCategorized|) (raw: string): Choice<struct (string * StreamId), unit> =
+            match tryParse raw with
+            | ValueSome catAndId -> Categorized catAndId
+            | ValueNone -> NotCategorized
 
-    (* Parsing: Raw Stream name Validation functions/pattern that handle malformed cases without throwing *)
+    let private throwInvalid raw = invalidArg "raw" (sprintf "Stream Name '%s' must contain a '-' separator" raw)
 
-    /// <summary>Validates and maps a trusted Stream Name consisting of a Category and an Id separated by a '-' (dash).<br/>
+    /// <summary>Validates and maps a Stream Name consisting of a Category and an StreamId separated by a '-' (dash).<br/>
     /// Throws <c>InvalidArgumentException</c> if it does not adhere to that form.</summary>
-    let parse (rawStreamName : string) : StreamName =
-        if rawStreamName.IndexOf '-' = -1 then
-            invalidArg "rawStreamName" (sprintf "Stream Name '%s' must contain a '-' separator" rawStreamName)
-        UMX.tag rawStreamName
-
-    let private dash = [|'-'|] // Separates {category}-{streamId}
-
-    /// <summary>Attempts to split a Stream Name in the form <c>{category}-{streamId}</c> into its two elements.
-    /// The <c>{streamId}</c> segment is permitted to include embedded '-' (dash) characters
-    /// Returns <c>None</c> if it does not adhere to that form.</summary>
-    let trySplitCategoryAndStreamId (rawStreamName : string) : struct (string * string) voption =
-        match rawStreamName.Split(dash, 2) with
-        | [| cat; id |] -> ValueSome struct (cat, id)
-        | _ -> ValueNone
+    let parse (raw: string): StreamName =
+        if raw.IndexOf Category.Separator = -1 then throwInvalid raw
+        raw |> Internal.trust
 
-    /// <summary>Attempts to split a Stream Name in the form <c>{category}-{streamId}</c> into its two elements.
-    /// The <c>{streamId}</c> segment is permitted to include embedded '-' (dash) characters
-    /// Yields <c>NotCategorized</c> if it does not adhere to that form.</summary>
-    let (|Categorized|NotCategorized|) (rawStreamName : string) : Choice<struct (string * string), unit> =
-        match trySplitCategoryAndStreamId rawStreamName with
-        | ValueSome catAndId -> Categorized catAndId
-        | ValueNone -> NotCategorized
-
-    (* Rendering *)
-
-    /// Strip off the strong typing (It's recommended to pattern match as below in the general case)
-    let inline toString (streamName : StreamName) : string =
-        UMX.untag streamName
+    /// Creates a StreamName in the canonical form; a category identifier and an streamId representing the aggregate's identity
+    /// category is separated from id by `-`
+    let create (category: string) (streamId: StreamId): StreamName =
```

Comment:
Will Equinox need to take a dep n FsCodec for this function now?

Reply:
looking at it atm - can either
- (tried so far) made Core depend on FsCodec (to get sid)
- move Category and Tracing into Equinox, making that available to anyone that depends only on Equinox (maybe splitting into a Stream.fs and Category.fs)

Atm I'm thinking that a StreamName.render will live in Core, and the stores will continue to work with just unadorned strings

Reply:
So Equinox has a dep on FsCodec now as Category is there and it needs StreamId
Dynamo and Cosmos also use the `category` function in metrics impl, but other than that, there's little manipulation of SN's within the stores - they just get it as a bare string